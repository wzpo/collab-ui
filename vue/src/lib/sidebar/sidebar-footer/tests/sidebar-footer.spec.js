import { mount, shallowMount } from '@vue/test-utils';
import SidebarFooter from '../index.vue';

describe('SidebarFooter', () => {
  it('should match snapshot', () => {
    const wrapper = mount(SidebarFooter);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should add customized class name if className prop is set', () => {
    const wrapper = shallowMount(SidebarFooter, {
      propsData: {
        class: 'testClassName'
      }
    });

    expect(wrapper.find('.testClassName').exists()).toBeTruthy();
  });

  it('should render children if children prop is set', () => {
    const wrapper = shallowMount(SidebarFooter, {
      slots: {
        default: `<div class='dummy-children'>Dummy Children</div>`
      }
    });

    expect(wrapper.find('.dummy-children').exists()).toBeTruthy();
  });
});

