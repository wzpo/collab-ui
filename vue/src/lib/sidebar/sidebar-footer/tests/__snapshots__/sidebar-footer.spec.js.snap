// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFooter should match snapshot 1`] = `<div class="md-sidebar__footer"></div>`;
